File: rust/lib/src/testsuite/test_context.rs
use core::panic;

use anyhow::Error;

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
pub struct TestContext {}

impl TestContext {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
    pub fn fatal(&self, err: Error) {
        TestContext::fail_test(err);
    }

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
    pub fn assert_true(&self, condition: bool, err: Error) {
        if !condition {
            TestContext::fail_test(err);
        }
    }

    fn fail_test(err: Error) {
        panic!("{:?}", err);
    }
}

File: golang/lib/testsuite/test_context.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package testsuite

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
type TestContext struct {}

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
func (context TestContext) Fatal(err error) {
	// We rely on panicking here because we want to completely stop whatever the test is doing
	failTest(err)
}

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
func (context TestContext) AssertTrue(condition bool, err error) {
	if (!condition) {
		failTest(err)
	}
}

func failTest(err error) {
	panic(err)
}
